### DISCUSSION TITLE
How to stop function  map.on 'click' add new marker on the map?

### DISCUSSION THREAD

**Discussion by hoangdangcodecomplete** [tomik23/react-leaflet-examples#11] [Q&A]
Hi, my friend!
I have implemented the code 1 function similar to your add move and delete marker function.
My difference is that I customize a button control and when I click on that control button, I will allow adding a new marker,  when I click cancel, it will stop the action of adding a marker. However, I cannot stop the marker click action on the map.
Do you have any solution to my problem?
If yes, please help me, thank you very much.
Video record about my issues: [Link demo](https://youtu.be/Kkfxk_HGjxc)

Here is my code: [Link code](https://github.com/hoangdangcodecomplete/DamageAssessmentMap/blob/main/button-control.jsx)

**Comment by tomickigrzegorz** [comment]
Hi,
I would set a variable to false  `x = false`.
Clicking on the icon sets the variable to `x = true` and we can add markers, clicking "cancel" sets the variable to `x = false` and blocks adding markers. Of course, you have to check the state of this variable every time.
This is probably the simplest solution.

**Comment by hoangdangcodecomplete** [comment]
<img width="600" alt="Screen Shot 2022-06-15 at 08 22 52" src="https://user-images.githubusercontent.com/104552361/173716405-2a79a62f-6b7e-492c-97df-a57b1b0a9b26.png">

Hi my friend,
Yes my idea was the same initially and I did it but now when I press cancel the x (in my code is isShowMarkerNote) variable is set to false however I can't stop the action of clicking on the map marker.

You can see I have useEffect and check every time that variable changes however preventing click action are not possible.
